gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -79,6 +82,34 @@
 @AutoService(FileSystemProvider.class)
 public final class CloudStorageFileSystemProvider extends FileSystemProvider {
 
+  private static class LazyPathIterator extends AbstractIterator<Path> {
+    private final Iterator<Blob> blobIterator;
+    private final Filter<? super Path> filter;
+    private final CloudStorageFileSystem fileSystem;
+
+    LazyPathIterator(CloudStorageFileSystem fileSystem, Iterator<Blob> blobIterator, Filter<? super Path> filter) {
+      this.blobIterator = blobIterator;
+      this.filter = filter;
+      this.fileSystem = fileSystem;
+    }
+
+    @Override
+    protected Path computeNext() {
+      while (blobIterator.hasNext()) {
+        Path path = fileSystem.getPath(blobIterator.next().name());
+        try {
+          if (filter.accept(path)) {
+            return path;
+          }
+        } catch (IOException ex) {
+          throw new DirectoryIteratorException(ex);
+        }
+      }
+      return endOfData();
+    }
+  }
+
+
   private final Storage storage;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2403961646/issues/source?bid=3123107&fileBranchId=3225960#l113)